app/models/pipeline_run.rb

```diff
@@ -160,31 +166,168 @@ def succeeded?
+    json_dict = JSON.parse(File.read(downloaded_json_path))
+
+    pipeline_output_dict = json_dict['pipeline_output']
+    pipeline_output_dict.slice!('remaining_reads', 'total_reads', 'taxon_counts_attributes')
+
+    self.total_reads = pipeline_output_dict['total_reads']
+    self.remaining_reads = pipeline_output_dict['remaining_reads']
+    self.unmapped_reads = count_unmapped_reads
+    self.fraction_subsampled = subsample_fraction
+
+    stats_array = JSON.parse(File.read(downloaded_stats_path))
+    stats_array = stats_array.select { |entry| entry.key?("task") }
+
+    version_s3_path = "#{alignment_output_s3_path}/#{VERSION_JSON_NAME}"
+    self.version = `aws s3 cp #{version_s3_path} -`
+
+    # only keep counts at certain taxonomic levels
+    taxon_counts_attributes_filtered = []
+    acceptable_tax_levels = [TaxonCount::TAX_LEVEL_SPECIES]
+    acceptable_tax_levels << TaxonCount::TAX_LEVEL_GENUS if multihit?
+    acceptable_tax_levels << TaxonCount::TAX_LEVEL_FAMILY if multihit?
+    pipeline_output_dict['taxon_counts_attributes'].each do |tcnt|
+      # TODO:  Better family support.
+      if acceptable_tax_levels.include?(tcnt['tax_level'].to_i) && !invalid_family_call?(tcnt)
+        taxon_counts_attributes_filtered << tcnt
+      end
+    end
+
+    job_stats.delete_all
+    self.job_stats_attributes = stats_array
+    self.taxon_counts_attributes = taxon_counts_attributes_filtered
+    self.updated_at = Time.now.utc
+    save
+    # aggregate the data at genus level
+    generate_aggregate_counts('genus') unless pr.multihit?
+    # merge more accurate name information from lineages table
+    update_names
+    # denormalize superkingdom_taxid into taxon_counts
+    if multihit?
+      update_superkingdoms
+    else
+      update_genera
+    end
+    # label taxa as phage or non-phage
+    update_is_phage
+
+    # rm the json
+    _stdout, _stderr, _status = Open3.capture3("rm -f #{downloaded_json_path} #{downloaded_stats_path}")
+  end
+
+  def db_load_postprocess
+    byteranges_json_s3_path = "#{postprocess_output_s3_path}/#{TAXID_BYTERANGE_JSON_NAME}"
+    downloaded_byteranges_path = PipelineRun.download_file(byteranges_json_s3_path, local_json_path)
+    taxon_byteranges_csv_file = "#{local_json_path}/taxon_byteranges"
+    hash_array_json2csv(downloaded_byteranges_path, taxon_byteranges_csv_file, %w[taxid hit_type first_byte last_byte])
+    ` cd #{local_json_path};
+      sed -e 's/$/,#{id}/' -i taxon_byteranges;
+      mysqlimport --replace --local --user=$DB_USERNAME --host=#{rds_host} --password=$DB_PASSWORD --columns=taxid,hit_type,first_byte,last_byte,pipeline_run_id --fields-terminated-by=',' idseq_#{Rails.env} taxon_byteranges;
+    `
+    _stdout, _stderr, _status = Open3.capture3("rm -f #{downloaded_byteranges_path}")
+  end
+
+  def s3_output_for(db_load_command)
+    case db_load_command
+    when "db_load_host_filtering"
+      "#{host_filter_output_s3_path}/#{STATS_JSON_NAME}"
```

Comment:
I think we should take opportunity to not make the result loading based on stats.json. 
Since we don't care about the stages any more.  I think we should reload stats.json constantly.  

For other files, we just need to load it once and we can make the result_monitor  
1. load stats.json constantly 
2. load the files we want whenever it's available. When all the files are loaded set the result_status to 'CHECKED'

Reply:
Ok. I will remove the logic that kills the job when `output_ready?` so we can set the target output in the web to anything we want without having to make sure it is the last file generated in idseq-pipeline. Instead, I will kill any zombie jobs when `results_completed?`, which means that ALL results are there.

Reply:
i thought we are getting rid of this?

Reply:
i think it's kinda problematic here to use stats.json for any stage. we should use a different file.

Reply:
Ok! I will use a reads_per_gene.tab (ercc counts)